### PR TITLE
Refactor: Extract review prompt eligibility into the shared module

### LINKED ISSUE #540: Review prompt eligibility logic is duplicated across platforms with no tests
The review prompt eligibility rules are hand-ported between Kotlin and Swift, with the four threshold constants written out twice:

- `app/src/main/java/com/baijum/ukufretboard/data/ReviewPromptRepository.kt` — `MIN_ACTIVE_DAYS`, `MIN_DAYS_SINCE_INSTALL`, `MAX_DISMISSALS`, `COOLDOWN_DAYS`
- `iosApp/UkuleleCompanion/Helpers/ReviewPromptManager.swift` — the same four in `private enum Constants`

`AGENTS.md` / `CLAUDE.md` is explicit that pure business logic belongs in `:shared`, and this qualifies: given a clock reading and a state snapshot, `isEligible()` is a pure function.

There are also **no tests on either platform** — `grep -r ReviewPrompt` across `app/src/test`, `app/src/androidTest`, `shared/src`, and the iOS test target returns nothing.

### This has already caused drift

The two implementations do not behave the same:

- iOS has `recordPromptShown()` (`ReviewPromptManager.swift:64`) called at display time. Android has no equivalent, so if the Android process dies with the dialog on screen, it re-prompts on next launch.
- Android shows a "thanks for the feedback" second step. iOS never does — `review_feedback_thanks` is translated into all 16 iOS locales in `Localizable.xcstrings` and is not referenced by any `.swift` file.

### Suggested fix

Extract the eligibility decision into `shared/src/commonMain/.../domain/` taking the clock and state as parameters, leave only the platform storage in `ReviewPromptRepository` / `ReviewPromptManager`, and add tests covering the date and count boundaries. That fixes the drift and the coverage gap together.

### PR DESCRIPTION
**Pull request by baijum**
Closes #540.

The five review-prompt eligibility gates were hand-ported between Kotlin and Swift with the four threshold constants written out twice, and neither copy had a single test. That combination is what let the two platforms drift apart in the first place.

## What changed

**New:** `shared/src/commonMain/.../domain/ReviewPromptEligibility.kt` — the decision as a pure function over a state snapshot and a clock reading, plus the four thresholds in one place.

**Android** (`ReviewPromptRepository.kt`) and **iOS** (`ReviewPromptManager.swift`) keep only the platform storage. Both build a `State` snapshot and delegate; neither names a threshold any more, so there is nothing left to drift. The `recordActiveDay` bound also goes through `shouldRecordActiveDay()` rather than each side comparing against its own copy of `MIN_ACTIVE_DAYS`.

Storage formats and keys are untouched — including the `dismiss_count` key kept for install compatibility — so no migration.

iOS persists these timestamps as epoch seconds and has since the first release, so it converts to milliseconds at the call boundary rather than rewriting what is already on disk.

## Tests

19 tests in `ReviewPromptEligibilityTest.kt` covering every gate and **both sides of each boundary**: exactly-at vs one millisecond before the install threshold and the cooldown, at vs below the prompt cap, at vs below the active-day minimum. Plus two cases the platform code had only ever handled implicitly:

- `lastPrompted == 0` means "never prompted", not "prompted at the epoch" — invisible with a present-day clock, so it is pinned with a clock close to zero.
- A clock that has moved backwards yields a negative elapsed span, which fails the gate rather than passing it.

Each test was verified to discriminate: the implementation was mutated (`>=` → `>`, `<` → `<=`, dropping the sentinel guard) and the expected test failed each time. The sentinel test exists *because* the first version of that assertion passed against the mutant.

## Verification

| Gate | Result |
|---|---|
| `:shared:allTests` | pass |
| `:app:testDebugUnitTest` | pass |
| `./gradlew detekt` | pass, no baseline change |
| `./gradlew lintDebug` | pass |
| `scripts/ktlint.sh --baseline=ktlint-baseline.xml` | pass, no baseline change |

The iOS side was type-checked against the generated framework (`swiftc -typecheck`, Swift 6, iOS 16 simulator) rather than built end to end — a full `xcodebuild` needs the 216 MB `onnxruntime.xcframework` that was removed from this worktree by request. That check did catch a real error: the nested Kotlin type bridges as `ReviewPromptEligibility.State`, not `ReviewPromptEligibilityState`. No Swift files were added, so `project.pbxproj` is untouched. CI runs the full iOS build.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Improvements**
  * Review prompts now use consistent eligibility rules across Android and iOS.
  * Active-day tracking, installation age, prompt limits, review status, and cooldown timing are handled consistently.
  * Existing prompt history and preferences remain supported.

* **Bug Fixes**
  * Improved handling of invalid or future timestamps and completed reviews.
  * Prevented duplicate active-day counting and prompts beyond configured limits.

* **Tests**
  * Added comprehensive Android, iOS, and shared coverage for review prompt eligibility and persistence.

### CODE REVIEW

File: iosApp/UkuleleCompanion/Helpers/ReviewPromptManager.swift [hunk 1]
```diff
@@ -76,33 +81,38 @@ final class ReviewPromptManager {
...
 
     // MARK: - Private
 
+    private func snapshot() -> ReviewPromptEligibility.State {
+        ReviewPromptEligibility.State(
+            activeDayCount: Int32(activeDaysCount()),
+            firstLaunchMillis: Self.millis(from: defaults.double(forKey: Keys.firstLaunch)),
+            hasReviewed: hasReviewed(),
+            promptCount: Int32(promptCount()),
+            lastPromptedMillis: Self.millis(from: defaults.double(forKey: Keys.lastPrompted))
```
[the remark this answers is not in the document]
Comment (baijum) [reply iosApp/UkuleleCompanion/Helpers/ReviewPromptManager.swift:105]: Fixed in 8d0c52a. Both counts now use `Int32(clamping:)`, and the shared rules reject a negative prompt count via a range test rather than `>= MAX_PROMPTS`, so a clamped `Int32.min` cannot slip past the cap.

Verified by mutation: reverting the clamp crashes the iOS test suite on the trap, and reverting the range test fails the new shared test.
